### PR TITLE
[WIP] CI Enforce View Props are Implemented

### PR DESCRIPTION
**Pull request by NickGerleman**
This begins to add tests which compare the native props a ViewManager reports to support against ViewConfigs kept in React Native. We begin with RCTView.

We fail to report a decent number of props that we do actually support, so we need to fix up some native code to report more. Still going through some of this right now. If all is looking correct, we will see error messages like the following in CI if we pull in a new version of RN without either implementing a property or adding it to an ignore list:

image

Note that this needs to reach into the internals of React Native for this type information. We also do this from Typescript, so we don't get break typings. This fragility is non-ideal, but should be okay for test code, and can be adapted if bits fail when bringing in new RN versions.

After this, we should:
- Add other native components
- Hook up babel-plugin-inline-view-configs to allow these checks for codegen'd types
- Compare VM commands against the ViewConfig

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/6098)

### CODE REVIEW

File: packages/IntegrationTest/tests/ViewManagerParityTests.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        opts.componentaName
+      }" is missing support for properties: ${unsupportedProps
+        .map(p => `\n - ${p}`)
+        .join('')}`,
+    );
+  }
+
+  const falseNyiProps = _.difference(
+    viewManagerConfigProps,
+    (opts.notYetImplementedProps || []).map(p => p[0]),
```
Comment (NickGerleman) [inline packages/IntegrationTest/tests/ViewManagerParityTests.ts:50]: This isn't right

File: packages/IntegrationTest/tests/ViewManagerParityTests.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+import {functionTest} from './lib/TestDefinition';
+import {UIManager} from 'react-native';
+
+const LayoutPropTypes = require('react-native/Libraries/DeprecatedPropTypes/DeprecatedLayoutPropTypes');
+const ReactNativeViewConfigRegistry = require('react-native/Libraries/Renderer/shims/ReactNativeViewConfigRegistry');
+
+const layoutProps = Object.keys(LayoutPropTypes);
+
+function diffViewManagerViewConfig(opts: {
+  componentaName: string;
```
Comment (NickGerleman) [inline packages/IntegrationTest/tests/ViewManagerParityTests.ts:18]: Typo

**Comment by TheSavior** [comment]
You may want to check out `verifyComponentAttributeEquivalence` which we use in core for this:

https://github.com/facebook/react-native/blob/c015f485c21057505cfb701a725d782d0b2fa220/Libraries/Utilities/verifyComponentAttributeEquivalence.js#L19-L42

**Comment by NickGerleman** [comment]
> You may want to check out `verifyComponentAttributeEquivalence` which we use in core for this:
> 
> https://github.com/facebook/react-native/blob/c015f485c21057505cfb701a725d782d0b2fa220/Libraries/Utilities/verifyComponentAttributeEquivalence.js#L19-L42

That is 100% where I stole the idea from 😉. The big difference is that we go the opposite direction, of checking that everything present in the JS view config is also present in the native view config. Starting this off with just existence of NativeProps/validAttributes for now.

**Comment by TheSavior** [comment]
> > You may want to check out `verifyComponentAttributeEquivalence` which we use in core for this:
> > https://github.com/facebook/react-native/blob/c015f485c21057505cfb701a725d782d0b2fa220/Libraries/Utilities/verifyComponentAttributeEquivalence.js#L19-L42
> 
> That is 100% where I stole the idea from 😉. The big difference is that we go the opposite direction, of checking that everything present in the JS view config is also present in the native view config. Starting this off with just existence of NativeProps/validAttributes for now.

Cool, glad you saw that. :D The reason we go the direction we do is because the JS view configs are a union of the props on both Android and iOS. Since the two native platforms have some differences that need to be resolved still. For Windows, it makes sense that the direction we are going wouldn't be helpful because it wouldn't catch missing things not implemented in native. However, it will push you to add all the iOS and Android divergent props and events as well. So you'll probably need to track what is the actual right thing to be implementing and maybe have a hardcoded list of props you can skip implementing in native.

File: packages/IntegrationTest/tests/ViewManagerParityTests.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+  const falseNyiProps = _.difference(
+    viewManagerConfigProps,
+    (opts.notYetImplementedProps || []).map(p => p[0]),
+  );
+
+  if (falseNyiProps.length > 0) {
+    throw new Error(
+      `The following props for ${
+        opts.componentaName
```
Comment (asklar) [inline packages/IntegrationTest/tests/ViewManagerParityTests.ts:56]: typo

**Comment by NickGerleman** [comment]
Not worth doing before Fabric.

**Pull request closed without merging**